tests/ProxyManagerTestAsset/RemoteProxy/RemoteServiceWithDefaultsInterface.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ProxyManagerTestAsset\RemoteProxy;
+
+/**
+ * Simple interface for a remote API that methods with default parameters
+ *
+ * @author Pedro Tanaka <pedro.stanaka@gmail.com>
+ * @license MIT
+ */
+interface RemoteServiceWithDefaultsInterface
```

Comment:
We need a data-provider entry with a method that contains more than one optional parameter, to verify that the intersection of optional and non-optional parameters works as expected.

For example, 2 required parameters, 3 optional parameters, 4 parameters provided, 5 parameters passed to client in total (of which one is a default)

Reply:
I added to the lang-feature tests, do you prefer that I add here as well?

Btw, the test I wrote with the variadic + optional parameters broke in the middle of the refactoring 🎊

Reply:
I think an integration test with a spy (mock) remote adapter is better suited for testing these added scenarios (the spy would expect a specific number of arguments)

Reply:
Okay. I added them now. I think it should be fine.